security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -298,12 +369,17 @@ func (sf *SecretFetcher) scrtDeleted(obj interface{}) {
 	}
 
 	rootCertResourceName := key + IngressGatewaySdsCaSuffix
-	// If there is root cert secret with the same resource name, delete that secret now.
-	sf.secrets.Delete(rootCertResourceName)
-	secretFetcherLog.Infof("secret %s is deleted", rootCertResourceName)
-	// Delete all cache entries that match the deleted key.
-	if sf.DeleteCache != nil {
-		sf.DeleteCache(rootCertResourceName)
+	rootSecret, exists := sf.secrets.Load(rootCertResourceName)
```

Comment:
why look up here before delete but no lookup for non-root secret above ?
```
key := scrt.GetName()
sf.secrets.Delete(key)
```

Reply:
We don't need to look up for non-root secret above.
The logic here is,
- when compound secret `bookinfo` is created, two secret items are added into secretfetcher's store, one is `bookinfo` which has server key/cert, the other is `bookinfo-cacert` which has client CA cert and its **RootCertOwnedByCompoundSecret** field is true. If operator runs kubectl delete secret `bookinfo`, both secret items should be deleted. There is no way to only delete the secret item `bookinfo-cacert` by operator.
- when client CA only secret `bookinfo-cacert` is created, one secret item is added into secretfetcher's store, which is `bookinfo-cacert` and its **RootCertOwnedByCompoundSecret** field is false. When operator deletes this secret. The item `bookinfo-cacert` will be deleted at line 364. Then it tries to find secret item `bookinfo-cacert-cacert` in the store at line 375, which could not exist.